code/game/gamemodes/gang/recaller.dm

```diff
@@ -0,0 +1,454 @@
+			if(ganger.current && (ganger.current.z <= 2) && (ganger.current.stat == CONSCIOUS))
+				ganger.current << ping
+		for(var/mob/M in dead_mob_list)
+			M << ping
+		log_game("[key_name(user)] Messaged [gang.name] Gang: [message].")
+
+
+/obj/item/device/gangtool/proc/register_device(mob/user)
+	if(gang)	//It's already been registered!
+		return
+	if((promotable && (user.mind in ticker.mode.get_gangsters())) || (user.mind in ticker.mode.get_gang_bosses()))
+		gang = user.mind.gang_datum
+		gang.gangtools += src
+		icon_state = "gangtool-[gang.color]"
+		if(!(user.mind in gang.bosses))
+			ticker.mode.remove_gangster(user.mind, 0, 2)
+			gang.bosses += user.mind
+			user.mind.gang_datum = gang
+			user.mind.special_role = "[gang.name] Gang Lieutenant"
+			gang.add_gang_hud(user.mind)
+			log_game("[key_name(user)] has been promoted to Lieutenant in the [gang.name] Gang")
+			free_pen = 1
+			gang.message_gangtools("[user] has been promoted to Lieutenant.")
+			user << "<FONT size=3 color=red><B>You have been promoted to Lieutenant!</B></FONT>"
+			ticker.mode.forge_gang_objectives(user.mind)
+			ticker.mode.greet_gang(user.mind,0)
+			user << "The <b>Gangtool</b> you registered will allow you to purchase weapons and equipment, and send messages to your gang."
+			user << "Unlike regular gangsters, you may use <b>recruitment pens</b> to add recruits to your gang. Use them on unsuspecting crew members to recruit them. Don't forget to get your one free pen from the gangtool."
+	else
+		usr << "<span class='warning'>ACCESS DENIED: Unauthorized user.</span>"
+
+/obj/item/device/gangtool/proc/recall(mob/user)
+	if(!can_use(user))
+		return 0
+
+	if(recalling)
+		usr << "<span class='warning'>Error: Recall already in progress.</span>"
+		return 0
+
+	gang.message_gangtools("[usr] is attempting to recall the emergency shuttle.")
+	recalling = 1
+	loc << "<span class='info'>\icon[src]Generating shuttle recall order with codes retrieved from last call signal...</span>"
+
+	sleep(rand(100,300))
+
+	if(shuttle_master.emergency.mode != SHUTTLE_CALL) //Shuttle can only be recalled when it's moving to the station
+		user << "<span class='warning'>\icon[src]Emergency shuttle cannot be recalled at this time.</span>"
+		recalling = 0
+		return 0
+	loc << "<span class='info'>\icon[src]Shuttle recall order generated. Accessing station long-range communication arrays...</span>"
+
+	sleep(rand(100,300))
+
+	if(!gang.dom_attempts)
+		user << "<span class='warning'>\icon[src]Error: Unable to access communication arrays. Firewall has logged our signature and is blocking all further attempts.</span>"
+		recalling = 0
+		return 0
+
+	var/turf/userturf = get_turf(user)
+	if(userturf.z != 1) //Shuttle can only be recalled while on station
+		user << "<span class='warning'>\icon[src]Error: Device out of range of station communication arrays.</span>"
+		recalling = 0
+		return 0
+	var/datum/station_state/end_state = new /datum/station_state()
```

Comment:
Okay.

* The type of this datum really hints that this should be a global thing...
* You should also move these station states into a new file because they're too hard to find.